api-tests/dev_apis/internal_trusted_storage/test_s006/test_s006.c

```diff
@@ -28,59 +28,56 @@
 
 client_test_t test_s006_sst_list[] = {
     NULL,
-    psa_sst_get_apis_check_for_different_flags_value,
+    psa_sst_flags_not_supported,
     NULL,
 };
 
 static uint8_t write_buff[TEST_BUFF_SIZE] = {
   0x00, 0x01, 0x02, 0x03, 0x04, 0x23, 0xF6, 0x07, 0x08, 0x0D, 0x0A, 0x1B, 0x0C, 0x5D, 0x0E,\
   0x70, 0xA1, 0xFF, 0xFF, 0x14, 0x73, 0x46, 0x97, 0xE8, 0xDD, 0xCA, 0x0B, 0x3C, 0x0D, 0x2E};
 
-static int32_t psa_sst_set_api_with_flag_value(psa_sst_uid_t uid, uint32_t data_len,
+static int32_t psa_sst_remove_api(psa_sst_uid_t uid, uint32_t data_len,
                                    uint8_t *data_buff, psa_sst_create_flags_t create_flag)
 {
     uint32_t status;
 
-    /* Set the uid with the data_len and data_buff */
-    status = SST_FUNCTION(s006_data[1].api, uid, data_len, data_buff, create_flag);
-    TEST_ASSERT_EQUAL(status, s006_data[1].status, TEST_CHECKPOINT_NUM(1));
-
-    /* Call the GET_INFO function and match the attributes */
+    /* Call the get_info function and match the attributes */
     status = SST_FUNCTION(s006_data[2].api, uid, &info);
     TEST_ASSERT_EQUAL(status, s006_data[2].status, TEST_CHECKPOINT_NUM(2));
     TEST_ASSERT_EQUAL(info.flags, create_flag, TEST_CHECKPOINT_NUM(3));
 
-    /* Remove the uid  */
+    /* Remove the UID  */
     status = SST_FUNCTION(s006_data[4].api, uid);
     TEST_ASSERT_EQUAL(status, s006_data[4].status, TEST_CHECKPOINT_NUM(4));
 
     return VAL_STATUS_SUCCESS;
 }
 
-int32_t psa_sst_get_apis_check_for_different_flags_value(security_t caller)
+int32_t psa_sst_flags_not_supported(security_t caller)
 {
    psa_sst_create_flags_t flag = 0x80000000;
    uint32_t status = VAL_STATUS_SUCCESS;
+   psa_sst_uid_t uid = UID_BASE_VALUE + 5;
 
-   /* Calling SET function with different create flag value */
+   /* Calling set function with different create flag value */
 
-   val->print(PRINT_TEST, "[Check 1] SET api call with valid flag values\n", 0);
+   val->print(PRINT_TEST, "[Check 1] set API call with valid flag values\n", 0);
    while (flag)
    {
-       if (psa_sst_set_api_with_flag_value(UID_BASE_VALUE, TEST_BUFF_SIZE, write_buff,
-                                          (flag & (~PSA_SST_FLAG_WRITE_ONCE))))
+       /* Create storage with flag value */
+       status = SST_FUNCTION(s006_data[1].api, uid, TEST_BUFF_SIZE, write_buff,
+                                                         (flag & (~PSA_SST_FLAG_WRITE_ONCE)));
+
+       if (status == s006_data[1].status)
        {
-           val->print(PRINT_ERROR, "Flag value =  %d\n", (flag & (~PSA_SST_FLAG_WRITE_ONCE)));
-           return VAL_STATUS_ERROR;
+          psa_sst_remove_api(uid, TEST_BUFF_SIZE, write_buff,
```

Comment:
We need to notify if this function fails.

Reply:
Comment taken care